dasharo-compatibility/device-detection.robot

```diff
@@ -0,0 +1,104 @@
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../pikvm-rest-api/pikvm_comm.robot
+Resource            ../lib/options/dcu.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+...                     AND
+...                     Set UEFI Option    UsbDriverStack    Enabled
+Suite Teardown      Run Keywords
+...                     Log Out And Close Connection
+...                     AND
+...                     Set UEFI Option    UsbDriverStack    Enabled
+
+
+*** Test Cases ***
+DDET001.001 USB Stack disable
+   [Documentation]    Test disabling the USB stack
+
+   Login To Linux With Root Privileges
+   Set UEFI Option    UsbDriverStack    Disabled
+
+   Execute Reboot Command
+   Sleep    10s
+
+   Set DUT Response Timeout    30s
+   Login To Linux With Root Privileges
+   ${out}=    Execute Command In Terminal
+   ...    cbmem -1 | grep "UsbBusStart:"
+
+   Should Contain    ${out}    usb bus started
+   ...    ignore_case=True
+
+DDET002.001 USB Stack enable
+   [Documentation]    Test enabling the USB stack
+
+   Login To Linux With Root Privileges
+   Set UEFI Option    UsbDriverStack    Enabled
+   Execute Reboot Command
+   Sleep    10s
+
+   Set DUT Response Timeout    30s
+   Login To Linux With Root Privileges
+   ${out}=    Execute Command In Terminal
+   ...    cbmem -1 | grep "UsbBusStart"
+
+   Should Contain    ${out}    usb bus started
+   ...    ignore_case=True
+
+DDET003.001 Usb Devices Detected In Firmware Warmboot
+    [Documentation]    Test if USB devices are detected after a warmboot
+
+    Login To Linux With Root Privileges
+    Perform Warmboot Using Rtcwake
+    Login To Linux With Root Privileges
+
+    ${out}=    Execute Command In Terminal
+    ...    cbmem -1 | grep -i 'UsbEnumeratePort'
+
+    Should Contain    ${out}    new device connected
+    ...    ignore_case=True
+
+DDET004.001 Usb Devices Detected In Firmware Suspension
+    [Documentation]    Test if USB devices are detected after a warmboot
+
+    Login To Linux With Root Privileges
+    Perform Suspend And Wake Using Rtcwake
+    Login To Linux With Root Privileges
+
+    ${out}=    Execute Command In Terminal
+    ...    cbmem -1 | grep -i 'UsbEnumeratePort'
```

Comment:
This log message is not generated on wake from suspend. This entire test case can be removed, because on the resume path from S3 we're skipping the entire UEFI payload which includes USB detection.

On wake from S0ix firmware is not actually involved at all

Reply:
Right, so I will just remove it as you said. https://github.com/Dasharo/open-source-firmware-validation/pull/344/commits/c45c746329bb97d1a22dd262b03002b6723f4961